### ISSUE TITLE
garden - keeps buying buildings so cant afford seeds

### ISSUE THREAD

**Issue by crashinz** [prinzstani/CookieBot#102]
I'm near the end of the game and have like 4 (non-shadow) achievements left

I still only have like 12 seeds, it never plants them. I assume because it cant ever afford them because it keeps buying buildings even though i don't have any achievements left requiring cookie count or having x amount of buildings etc.. (increasing cps just keeps increasing seed cost)

does it finally just cheat them in?

one of its current tasks is "Missing achievement #379: Harvest 1000 mature garden plants, try to get it now." but it hasn't planted any seeds

its only harvested like 261 plants, lvl 10 farm. little over a year playtime.

ive just got some shadow achievements, the last stock achievement, and garden achievements left

otherwise, really great, its done wonderful.

**Comment by prinzstani** [comment]
I see the problem.
The bot is not optimized to handle the garden late in the game.
When you let it run a few days, it will have enough buildings such that the buying of buildings does not get in the way of the garden. Typically, you should be in this situation after 2 days. Then you will have full speed in the garden and get all the achievements.
Achievement 379 does not have any specific code; the bot just does normal garden and building processing.
It will also do normal stock processing, so you might even get the stock achievement along the way.

This is also related to [issue   ##59](https://github.com/prinzstani/CookieBot/issues/59).

If the garden does not work after 5 days, please send an export save.

**Comment by Kikithecat12345** [comment]
Would it be possible to add spending limits (as a percentage of CPS per some timeframe, or something) so that if the bot is exclusively going for something that requires a CPS dependent amount of cookies, it can actually save up enough to make it work? 

A potential way this could be implemented is effectively adding a method or function that when called, sets the limit, and returns a function that can be called to release the limit, effectively allowing other parts of the code to make sure it doesn't get unlocked when it's expecting spending to be nerfed. (probably overcomplicated but I'm just throwing ideas at the wall.)

**Comment by PrinzAndreas** [comment]
Yes, that would be one way of doing it. However, as the bot will finish the garden anyway, just taking a few extra days, I hesitate to put in a lot of code to make it work. For me, this is related to the manual mode and will be solved there  (issue #59).
